crates/sui-protocol-config/src/lib.rs

```diff
@@ -511,4 +530,43 @@ impl ProtocolConfig {
             _ => panic!("unsupported version {:?}", version),
         }
     }
+
+    /// Override one or more settings in the config, for testing.
+    /// This must be called at the beginning of the test, before get_for_(min|max)_version is
+    /// called, since those functions cache their return value.
+    pub fn apply_overrides_for_testing(
+        override_fn: impl Fn(ProtocolVersion, Self) -> Self + Send + 'static,
+    ) -> OverrideGuard {
+        CONFIG_OVERRIDE.with(|ovr| {
+            let mut cur = ovr.lock().unwrap();
+            assert!(cur.is_none(), "config override already present");
+            *cur = Some(Box::new(override_fn));
+            OverrideGuard
+        })
+    }
+}
+
+// Setters for tests
+impl ProtocolConfig {
+    pub fn set_max_function_definitions_for_testing(&mut self, m: usize) {
```

Comment:
is the idea that `ProtocolConfig` will have all setters as well?

Reply:
anything you want to change for a test, you can add a setter here. I didn't want to add setters for everything since I assume 99% of these will never be set in a test.

Reply:
got it, I guess I am unclear whether it is a good model to have setters that will pop up everywhere, as opposed to a ctor that returns a given config. I mention only for us to think about (well you mostly :) ) not a blocker at all